dockstore-webservice/src/main/java/io/dockstore/webservice/resources/proposedGA4GH/ToolsApiExtendedServiceImpl.java

```diff
@@ -439,15 +435,43 @@ public Response submitMetricsData(String id, String versionId, Partner platform,
         try {
             ObjectMapper mapper = new ObjectMapper();
             String metricsData = mapper.writeValueAsString(executions);
-            // What format should metrics data be in? String or schema?
+            MetricsDataS3Client metricsDataS3Client = new MetricsDataS3Client(metricsConfig.getS3BucketName(), metricsConfig.getS3EndpointOverride());
             metricsDataS3Client.createS3Object(id, versionId, platform.name(), S3ClientHelper.createFileName(), owner.getId(), description, metricsData);
             return Response.noContent().build();
-        } catch (AwsServiceException | SdkClientException | JsonProcessingException e) {
+        } catch (AwsServiceException | SdkClientException | JsonProcessingException | URISyntaxException e) {
             LOG.error("Could not submit metrics data", e);
             throw new CustomWebApplicationException("Could not submit metrics data", HttpStatus.SC_BAD_REQUEST);
         }
     }
 
+    @Override
+    public Response setAggregatedMetrics(String id, String versionId, Partner platform, Metrics aggregatedMetrics) {
+        // Check that the entry and version exists
+        Entry<?, ?> entry;
+        try {
+            entry = getEntry(id, Optional.empty());
+        } catch (UnsupportedEncodingException | IllegalArgumentException e) {
+            return BAD_DECODE_REGISTRY_RESPONSE;
+        }
+
+        if (entry == null) {
+            return Response.status(Response.Status.NOT_FOUND.getStatusCode(), TOOL_NOT_FOUND_ERROR).build();
+        }
+
+        Version<?> version = getVersion(entry, versionId).orElse(null);
+        if (version == null) {
+            return Response.status(Response.Status.NOT_FOUND.getStatusCode(), VERSION_NOT_FOUND_ERROR).build();
+        }
+
+        // Check that the aggregated metrics have at least ExecutionStatusCount
+        if (aggregatedMetrics.getExecutionStatusCount() == null) {
+            return Response.status(Response.Status.BAD_REQUEST.getStatusCode(), EXECUTION_STATUS_COUNT_ERROR).build();
+        }
+
+        version.getMetricsByPlatform().put(platform, aggregatedMetrics);
```

Comment:
This got me thinking, are you able to verify that when you put new metrics, that the old metrics are deleted from the DB? The cascading/join stuff in Hibernate is a little confusing, to me at least, and we've had at least one bug where orphans were left over.

Reply:
I added this check in this commit https://github.com/dockstore/dockstore/pull/5381/commits/bfd0bc2e3ec89a12c7239aa087479e36f70f652f. Let me know if that's enough or if there's something else you had in mind